### DISCUSSION TITLE
rpc error: code = Unknown desc = NOAUTH Authentication required

### DISCUSSION THREAD

**Discussion by Knightly8889** [argoproj/argo-cd#20027] [General]
Hi Team, I am trying to create an application, but getting this error 
```
Unable to create application: application spec for argocd-main is invalid: InvalidSpecError: Unable to generate manifests in Code: rpc error: code = Unknown desc = NOAUTH Authentication required.
```

I am able to add the repository using

argocd repo add https://github.com --username   --password 

but still facing the same error. 
<img width="390" alt="Screenshot 2024-09-20 at 8 48 51 AM" src="https://github.com/user-attachments/assets/5d293744-50e1-4d67-ace3-08ae50c2a0d9">

**Comment by pasha-codefresh** [comment]
This issue should be resolved by this PR 
https://github.com/argoproj/argo-cd/pull/19599
